### PR TITLE
[CBW-1071] fixes for DTS id provider.

### PR DESCRIPTION
**Pull request by czerwix**
## Changes

[CBW-1071] fixes for DTS id provider.
## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

[CBW-1071]: https://concordium.atlassian.net/browse/CBW-1071?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/concordium/wallet/ui/identity/identityproviderwebview/IdentityProviderWebViewViewModel.kt [hunk 1]
```diff
@@ -220,12 +220,12 @@ class IdentityProviderWebViewViewModel(application: Application) : AndroidViewMo
                     override fun onResponse(call: Call<String>, response: Response<String>) {
                         val response = response.errorBody()?.string()
                         when {
-                            response?.contains(CreateIdentityError.NONE.message) == true -> {
+                            CreateIdentityError.NONE.containsStatusMessage(response) -> {
```
Comment (orhoj) [inline app/src/main/java/com/concordium/wallet/ui/identity/identityproviderwebview/IdentityProviderWebViewViewModel.kt:223]: To me this reads a little weird. When you have:
```
CreateIdentityError.NONE.containsStatusMessage()
```
then it sounds like that you are checking if the `ENUM` value `NONE` contains some status message, but we are actually checking the `response`.

**Comment by orhoj** [comment]
When navigating to the recovery page, then if I select **Cancel** when it asks for the pincode, then I stay on the blank recovery page. I guess it should return to where we came from instead.

**Comment by orhoj** [comment]
The error message shown for a duplicate `idCredPub` with Notabene is different from when getting it from DTS. I would assume this should be identical.

File: app/src/main/java/com/concordium/wallet/ui/identity/identityproviderwebview/IdentityProviderWebViewViewModel.kt (hunk 1 above)
Comment (czerwix) [reply app/src/main/java/com/concordium/wallet/ui/identity/identityproviderwebview/IdentityProviderWebViewViewModel.kt:223]: changed to 
image

**Comment by czerwix** [comment]
both issues should be fixed now

**Review by orhoj — APPROVED** [review]

**Pull request merged**